src/AutoRest.CSharp/Mgmt/Models/MgmtClientOperation.cs

```diff
@@ -85,18 +85,44 @@ private MgmtClientOperation(IReadOnlyList<MgmtRestOperation> operations, Formatt
         public MgmtRestOperation this[int index] => _operations[index];
 
         private MethodSignature? _methodSignature;
-        public MethodSignature MethodSignature => _methodSignature ??= new MethodSignature(
-            Name,
-            null,
-            Description,
-            Accessibility == Public
-                ? _extensionParameter != null
-                    ? Public | Static | Extension
-                    : Public | Virtual
-                : Accessibility,
-            IsPagingOperation
-                ? new CSharpType(typeof(Pageable<>), ReturnType)
-                : ReturnType, null, MethodParameters.ToArray());
+        public MethodSignature MethodSignature
+        {
+            get
+            {
+                if (_methodSignature != null)
+                    return _methodSignature;
+                else
+                {
+                    var attris = this._operations
+                        .Where(op => Configuration.MgmtConfiguration.PrivilegedOperations.ContainsKey(op.OperationId))
+                        .Select(op =>
+                        {
+                            var arg = Configuration.MgmtConfiguration.PrivilegedOperations[op.OperationId];
+                            MgmtReport.Instance.TransformSection.AddTransformLog(
+                                new TransformItem(TransformTypeName.PrivilegedOperations, op.OperationId, arg),
+                                op.Operation.GetFullSerializedName(),
+                                $"Operation {op.OperationId} is marked as Privileged Operation");
+                            return new CSharpAttribute(typeof(Azure.Core.CallerShouldAuditAttribute), arg);
+                        })
+                        .ToList();
```

Comment:
I believe all these content could be calculated when this `MgmtClientOperation` instance is created.
If we do in that way, we could significantly simply the code here.

Reply:
such as we add a field:
```
private readonly IReadOnlyList<CSharpAttribute> _attributes;

// intialize this in the ctor by calling a static method
```
and we could have this property almost unchanged, just add a `_attributes` argument.